core/src/com/unciv/logic/battle/GreatGeneralImplementation.kt

```diff
@@ -0,0 +1,141 @@
+                        && (tile.civilianUnit == null || tile.civilianUnit == general)
+                        && militaryUnit.getMaxMovement() <= unitMaxMovement
+                        && !tile.isCityCenter()
+            }
+
+        // rank tiles and find best 
+        val unitBonusRadius = generalBonusData.maxOfOrNull { it.radius }
+            ?: return null
+        return militaryUnitTilesInDistance
+            .maxByOrNull { unitTile ->
+                unitTile.getTilesInDistance(unitBonusRadius).sumOf { auraTile ->
+                    val militaryUnit = auraTile.militaryUnit
+                    if (militaryUnit == null || militaryUnit.civInfo != general.civInfo) 0
+                    else generalBonusData.firstOrNull {
+                        auraTile.aerialDistanceTo(unitTile) <= it.radius
+                                && (it.filter == "All" || militaryUnit.matchesFilter(it.filter))
+                    }?.bonus ?: 0
+                }
+            }
+    }
+
+    /** Find and cache the maximum Great General _Radius_ for a [civInfo]'s _BaseUnits_ */
+    fun setMaxGeneralBonusRadiusBase(civInfo: CivilizationInfo) {
+        civInfo.maxGeneralBonusRadiusBase = civInfo.gameInfo.ruleSet.units.values.asSequence()
+            .filter { it.isGreatGeneral() &&
+                    (it.uniqueTo == civInfo.civName || it.uniqueTo == null && civInfo.getEquivalentUnit(it) == it)
+            }.flatMap {
+                it.getMatchingUniques(UniqueType.GreatGeneralAura) +
+                        it.getMatchingUniques(UniqueType.BonusForUnitsInRadius)
+            }.maxOfOrNull {
+                if (it.type == UniqueType.BonusForUnitsInRadius) 2
+                else it.params[2].toIntOrNull() ?: 0
+            } ?: 0
+        updateMaxGeneralBonusRadius(civInfo)
+    }
+
+    /** Update [CivilizationInfo.maxGeneralBonusRadius] from [CivilizationInfo.maxGeneralBonusRadiusBase]
+     *  and all units having a promotion granting Great General abilities. Should be called whenever
+     *  a unit is destroyed or otherwise removed from the civ, or a unit gains a promotion. */
+    fun updateMaxGeneralBonusRadius(civInfo: CivilizationInfo) {
+        // If there are no promotions granting a Great General power, then we don't need to scan the actual MapUnits
+        val promotionsGrantingGeneralBonus = civInfo.gameInfo.promotionsGrantingGeneralBonus
+        if (promotionsGrantingGeneralBonus.isEmpty()) {
+            civInfo.maxGeneralBonusRadius = civInfo.maxGeneralBonusRadiusBase
+            return
+        }
+
+        // Need to scan the units - might be faster looping getCivUnits only once, but far less readable
+        val maxPromotionGeneralBonusRadius = promotionsGrantingGeneralBonus.asSequence()
+            .filter { promotion ->
+                civInfo.getCivUnits().any { unit->
+                    promotion.name in unit.promotions.promotions
+                }
```

Comment:
As far as I see "GG Aura" is a unique of the `Unit` type, not `Promotion` type. Why do we looking through the promotions here?

Reply:
Because `UniqueTarget.Promotion(Unit)` - they _are_ found and _do_ work from promotions. I didn't invest time in checking the implications of the "Aura" on UnitType - yea, `setMaxGreatGeneralBonusRadiusBase` should check them. Thanks!